api/src/main/java/com/datatorrent/api/LocalMode.java

```diff
@@ -18,34 +18,64 @@
  */
 package com.datatorrent.api;
 
-import java.util.Iterator;
-import java.util.ServiceLoader;
-
+import org.apache.apex.api.Launcher;
 import org.apache.hadoop.conf.Configuration;
 
+import com.google.common.base.Throwables;
+
 /**
  * Local mode execution for single application
  *
+ * @deprecated
  * @since 0.3.2
  */
-public abstract class LocalMode
+@Deprecated
+public abstract class LocalMode<H extends Launcher.AppHandle> extends Launcher<H>
 {
 
   /**
+   * Parameter to specify the time after which the application will be shutdown; pass 0 to run indefinitely.
+   */
+  public static final Attribute<Long> RUN_MILLIS = new Attribute<Long>(0L);
+
+  /**
+   * Parameter to launch application asynchronously and return from launch immediately.
+   */
+  public static final Attribute<Boolean> RUN_ASYNC = new Attribute<Boolean>(false);
+
+  /**
+   * Parameter to enable or disable heartbeat monitoring.
+   */
+  public static final Attribute<Boolean> HEARTBEAT_MONITORING = new Attribute<Boolean>(true);
+
+  /**
+   * Parameter to serialize DAG before launch.
+   */
+  public static final Attribute<Boolean> SERIALIZE_DAG = new Attribute<Boolean>(false);
+
+  static {
+    Attribute.AttributeMap.AttributeInitializer.initialize(LocalMode.class);
+  }
+
+  /**
    * <p>
    * getDAG.</p>
+   * @Deprecated
    *
    * @return
    */
+  @Deprecated
```

Comment:
In my IDE I see that if the method itself is not deprecated, calls to it don't have a strikethrough that deprecated methods have (only the instantiation has it if class is deprecated). I am seeing what is being suggested by the second responder in this stackoverflow discussions
http://stackoverflow.com/questions/17615019/class-with-deprecated-annotation-means-that-all-methods-and-fields-automaticall

I am open to changing it to just the class having deprecated. Let me know what you think based on the above.